### PR TITLE
API Changes & Tests for Property, ObjectSchema, & Schema

### PR DESCRIPTION
**Pull request by segiddins**
\c @alazier @jpsim

### CODE REVIEW

File: RealmSwift/Tests/ObjectSchemaTests.swift [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        let propertyNames = objectSchema.properties.map { $0.name }
+        XCTAssertEqual(propertyNames, ["boolCol", "intCol", "floatCol", "doubleCol", "stringCol", "binaryCol", "dateCol", "objectCol", "arrayCol"])
+    }
+
+    // Cannot name testClassName() because it interferes with the method on XCTest
+    func testClassNameProperty() {
+        XCTAssertEqual(objectSchema.className, "SwiftObject")
+    }
+
+    func testPrimaryKeyProperty() {
```
Comment (alazier) [inline RealmSwift/Tests/ObjectSchemaTests.swift]: should test for object with and without primary key

Comment (segiddins) [inline RealmSwift/Tests/ObjectSchemaTests.swift]: already do

File: RealmSwift/Tests/PropertyTests.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        XCTAssertEqual(primaryProperty.name, "stringCol")
+    }
+
+    func testType() {
+        XCTAssertEqual(primitiveProperty.type, PropertyType.Int)
+        XCTAssertEqual(linkProperty.type, PropertyType.Object)
+        XCTAssertEqual(primaryProperty.type, PropertyType.String)
+    }
+
+    func testIndexed() {
```
Comment (jpsim) [inline RealmSwift/Tests/PropertyTests.swift]: `class func indexedProperties() -> [String]` isn't being tested, only `primaryKey()`.

Comment (segiddins) [inline RealmSwift/Tests/PropertyTests.swift]: That's tested in the `ObjectSchemaInitializationTests`

Comment (jpsim) [inline RealmSwift/Tests/PropertyTests.swift]: Cool.

**Comment by jpsim** [comment]
:+1:

**Pull request merged**